js/process.ts

```diff
@@ -179,3 +193,37 @@ export function run(opt: RunOptions): Process {
 
   return new Process(res);
 }
+
+export enum Signal {
+  SIGHUP = 1,
+  SIGINT,
+  SIGQUIT,
```

Comment:
Although it's equivalent, I would prefer explicitly assigning values to each of these. It just makes things a bit more obvious if it ever needs to be debugged.
```js
SIGINT = 2,
SIGQUIT = 3,
/* .. */
```

Reply:
Actually I've just realized that certain signal numbers are platform dependent (like 10 is `SIGBUS` on macOS but `SIGUSR1` on Linux). I might also need to check the current platform to decide which set of signals to export.

Reply:
Ah too bad. I thought maybe posix defined those - [apparently not](https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/).
I think there's two options:
1. Do something similar to Deno.platform where we generate the typescript during build.
2. Pass the signals as strings ("SIGINT") and map them to their platform dependent integers outside of V8. This is what we did in Node IIRC.

Reply:
Currently resorting to
```ts
export const Signal = platform.os === "mac" ? MacOSSignal : LinuxSignal;
```

Using string names also not quite ideal unless we build a similar mapping on the Rust side, since [`from_str` in `nix` crate assumes mostly linux signal name mappings](https://github.com/nix-rust/nix/blob/80e716e41e2ce991a9ffc3a726c246f00f7e449d/src/sys/signal.rs#L24-L63)

Reply:
Ok that works